flask_user/views.py

```diff
@@ -590,9 +591,7 @@ def reset_password(token):
 
         # Change password
         hashed_password = user_manager.hash_password(form.new_password.data)
-        user_auth = user.user_auth if db_adapter.UserAuthClass and hasattr(user, 'user_auth') else user
-        db_adapter.update_object(user_auth, password=hashed_password)
-        db_adapter.commit()
+        user_manager.update_password(user, hashed_password)
```

Comment:
This change also minimizes the assumption that `password` is a field on the user entity. It also makes the code more consistent I think because `update_password` is already a method on `UserManager` so I don't know why it was not being used here (even though it has the same logic).